src/hooks/useAuthenticatedUser.ts

```diff
@@ -0,0 +1,16 @@
+import { useContext } from 'react'
+import { AuthContext } from '../context/auth'
+
+import { IAuthContext } from '../interfaces/IAuth'
+
+function useAuthenticatedUser(): IAuthContext {
+  const context = useContext(AuthContext)
+
+  if (context === undefined) {
```

Comment:
you can change for if(!!context)

this a way to transform context in boolean

Reply:
Nice! I will check this

Reply:
`!!context` is opposite to `context === undefined`, it's actually equivalent to `context !== undefined`, since `undefined` is falsy, it will negate once then twice, resulting in a truthy value. 

The correct way if you want to use this notation would be `!!!context`, but IMHO `context === undefined` is cleaner and more specific :)